drivers/i2c/i2c_ll_stm32.c

```diff
@@ -74,8 +74,6 @@ int i2c_stm32_runtime_configure(const struct device *dev, uint32_t config)
 
 #ifdef CONFIG_PM_DEVICE_RUNTIME
 	(void)pm_device_runtime_get(dev);
-#else
-	pm_device_busy_set(dev);
```

Comment:
> whether a SoC/core goes in to LP-mode would be dictated the timings provided in the DT power-state properties, depending on the actions done here, this may still occur.

Right, but for it to go in LP-mode, it has to be entering idle, which can't be the case until this function call is completed.

Reply:
Idle/LP-mode may be triggered from any deeper nested code waiting for a longer period than the residency + exit latency of a LP-mode. I saw this with the calibration steps in the ADC configuration logic.

As the STM32 maintainer, I'd defer to you if you guarantee this won't happen; just wanted to provide a heads-up :)

Reply:
> Idle/LP-mode may be triggered from any deeper nested code waiting for a longer period than the residency + exit latency of a LP-mode.

Agreed, but I don't see how it can happen if there is still code to execute. Even if you get interrupted by a ahigh prio task that complete and yield, current function execution will be resumed before the kernel can trigger idle.
If this isn't the case, well, I think we'll have much wide and critical issues to deal with.

Reply:
True, idle will only happen if some code inside waits for any reason. If that's not the case, then there's no purpose for the busy_set logic and you can safely ignore my warning above :)

My though process was that if it was there before it must have some concrete reason, e.g. the I2C configure code waiting for something long enough, but looking at v1/v2 `stm32_i2c_configure_timing()` that doesn't seem to be the case.